### DISCUSSION TITLE
How to train my images dataset with crnn_vgg16_bn

### DISCUSSION THREAD

**Discussion by Shiva-Ai4m** [mindee/doctr#1250] [General]
I wwant to train my image dataset. How can I train it? 
Can you please explain the steps , @felixT2K ?

**Comment by felixT2K** [comment]
Hi @Shivashankarar 👋🏼 ,

Please do not open further discussions with the same question. I have already replied to this in https://github.com/mindee/doctr/discussions/1247#discussioncomment-6400624 :)

Please take a look at the data format section: https://github.com/mindee/doctr/tree/main/references/recognition#data-format
Each image should contain only 1 word for example: https://www.mdpi.com/sensors/sensors-22-06025/article_deploy/html/images/sensors-22-06025-g016.png

Let's say the image with 0.0019 is `img_1.jpg ` in a `images` folder than the corresponding `labels.json` would look like:

```
{
    "img_1.jpg": "I",
}
```

Overall structure would be:
train_dataset/
    images  <-- contains all images
    labels.json   <-- contains all labels
    
 (same for the validation dataset)
 
 After your dataset is prepared you can run:
 `python3 references/recognition/train_pytorch.py crnn_vgg16_bn --train_path=<PATH_TO_TRAIN_SET> --val_path=<PATH_TO_VAL_SET> --pretrained`
